components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -834,10 +976,12 @@ public static void startTenantFlow(String tenantDomain, String userName)
 
     public static void startTenantFlow(String tenantDomain) throws IdentityApplicationManagementException {
 
-        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);;
+        String userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
         PrivilegedCarbonContext.startTenantFlow();
         PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
         PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUserId(userId);
```

Comment:
why do we need to set the user id here?

Reply:
When a new tenant flow is started the userId cannot be accessed in that flow.